### PR TITLE
Inform user that their camera is starting in Lobby

### PR DESCRIPTION
**Pull request by hughns**
Instead of just showing a grey box.

image

n.b. it is expected that this will be visible for a second or so whilst the browser and OS set up access to the camera.

### CODE REVIEW

**Comment by fkwp** [comment]
From internal discussion: I think text sounds apologetic. Maybe we can have a little spinner instead?

**Comment by americanrefugee** [comment]
Maybe show the avatar of the room you're about to join. Also consider adding a small spinner icon next to / above the message "Camera is starting".

**Comment by toger5** [comment]
Toggling the camera in the lobby it now looks like the avatar is appearing disappearing which i think is very nice feedback to switch from muted -> video.
I think putting a spinner in the center of the previous avatar would look really good.

**Comment by americanrefugee** [comment]
[How about this?](https://www.figma.com/design/fq7cTHPNHZX3ttFW1gBEnv/Compound-Element-Call?node-id=3751-7739&t=Su4bHfSG2373vHCv-4)

Loading video

Also, I changed my mind. The avatar should be YOUR avatar, not the room.

- The room avatar is already in the header
- The lobby is a preview of you, not the call

**Comment by toger5** [comment]
@americanrefugee does it look better with the greyed out image?
I think just the loader and the label would be enough?

**Comment by americanrefugee** [comment]
> @americanrefugee does it look better with the greyed out image? I think just the loader and the label would be enough?

Yes, it should be grayed out (50% opacity). Otherwise it will hard/impossible to see the spinner against the image - especially if the user avatar is light/white.

**Comment by hughns** [comment]
Benchmarking other applications:

Meet is showing the "Camera is starting" label:
image

Teams just goes to a grey tile until the video loads (a bit like what we have now):
image

Zoom doesn't provide feedback in the video tile (it continues to show the avatar), but has three states on the video button:

Off: image

Turning on: it's a grey intermediate version

On: image

**Comment by hughns** [comment]
The avatar is currently so large that you can't fit text below it at most resolutions.

The colour in this is `text-secondary` with `space-10x` below the top:

image

I will try a spinner on top of the avatar next.

**Comment by hughns** [comment]
It now looks like this which is an approximation of comments above:

image

**Comment by fkwp** [comment]
fine for me

File: src/Avatar.module.css [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  opacity: 0.5;
+  color: var(--cpd-color-fg-primary);
```
Comment (robintown) [inline src/Avatar.module.css]: This token doesn't exist (though it should, and hopefully will next time by the next time I look into tokens with Aaron: https://github.com/element-hq/compound/issues/313)

File: src/room/VideoPreview.tsx [hunk 2]
```diff
@@ -53,6 +55,11 @@ export const VideoPreview: FC<Props> = ({
     };
   }, [videoTrack]);
 
+  const cameraIsStarting = useMemo(
+    () => muteStates.video.enabled && (!videoTrack || !videoEl.current),
+    [muteStates.video.enabled, videoTrack, videoEl],
+  );
```
Comment (robintown) [inline src/room/VideoPreview.tsx]: The current value of a ref isn't generally something we can consume reactively. Maybe `videoEl.current` doesn't need to be part of the condition at all?

File: src/Avatar.tsx [hunk 3]
```diff
@@ -123,14 +129,23 @@ export const Avatar: FC<Props> = ({
...
-      size={`${sizePx}px`}
-      src={avatarUrl}
-      style={style}
-      {...props}
-    />
+    <div>
+      {loading && (
+        <div className={styles.loading}>
+          <InlineSpinner
+            size={typeof sizePx === "number" ? sizePx / 3 : undefined}
```
Comment (robintown) [inline src/Avatar.tsx]: If this loading indicator is designed to work with `size` being a number, maybe it's better to extract this bit from the Avatar component where it's more reasonable for it to be an ad-hoc thing. Then we could avoid the extra `<div>` wrapper being added to all avatars as well.

**Comment by hughns** [comment]
I've done a refactor based on @robintown's feedback. My intention is that the design will be used within a call as well. But this will go in another PR.

**Review by toger5 — APPROVED** [review]
The code looks good. 
I am not sure if we can do anything about the character in the span not being 100% centered...
That is a bit annoying. not a merge blocker but sth one could try in a follow up PR.
Proabably the Span needs to become a div -> and the span needs to be added a child to the diff with some custom margins.

**Pull request merged**